src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -78,7 +78,11 @@ public RepoConfiguration(String location, String branch) throws InvalidLocationE
     public RepoConfiguration(String location, String branch, List<String> formats, List<String> ignoreGlobList,
             boolean isStandaloneConfigIgnored, List<String> ignoreCommitList) throws InvalidLocationException {
         this.location = location;
-        this.branch = branch;
+        if (!location.isEmpty()) {
+            this.branch = branch;
+        } else {
+            this.branch = "";
+        }
```

Comment:
Can use ternary expressions to improve readability.

`this.branch = (!location.isEmpty()) ? branch : "";`